docs/operations/requests-mirroring-to-secondary-cluster.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Requests mirroring to secondary cluster"
+linkTitle: "Requests mirroring to secondary cluster"
+weight: 4
+slug: requests-mirroring-to-secondary-cluster
+---
+
+Requests mirroring (or shadowing) is a technique you can use to mirror requests from a primary Cortex cluster to a secondary one.
+
+For example, requests mirroring can be used when you need to setup a testing Cortex cluster receiving the same series ingested by a primary one without having control over Prometheus remote write config (if you do, then configuring two remote write entries in Prometheus would be the preferred option).
+
+## Mirroring with Envoy proxy
+
+[Envoy proxy](https://www.envoyproxy.io/) can be used to mirror HTTP requests to a secondary upstream cluster. From a network path perspective, you should run Envoy in front of both clusters distributors, letting Envoy to proxy requests to the primary Cortex cluster and mirror them to a secondary cluster in background. The performances and availability of the secondary cluster have no impact on the requests to the primary one. The response to the client will always be the one from the primary one. In this sense, the requests from Envoy to the secondary cluster are "fire and forget".
+
+### Example Envoy config
+
+The following Envoy configuration shows an example with two Cortex clusters. Envoy will listen on port `9900` and will proxies all requests to `cortex-primary:80`, mirroring it to `cortex-secondary:80` too.
+
+[embedmd]:# (./requests-mirroring-envoy.yaml)
+```yaml
```

Comment:
Isn't using both embedmd and snippet going to generate a duplicate snippet?

Reply:
embedmd does update the markdown file in place (that's what happen every time you type `make doc`) and is smart enough to update the snippet inside the markdown without generating duplicates.

Reply:
Thanks for explanation.